crates/llms/src/anthropic/types.rs

```diff
@@ -296,83 +296,44 @@ pub struct MetadataParam {
-        match s {
-            "claude-3-5-sonnet-20240620" => Ok(AnthropicModelVariant::Claude35Sonnet20240620),
-            "claude-3-opus-20240229" => Ok(AnthropicModelVariant::Claude3Opus20240229),
-            "claude-3-sonnet-20240229" => Ok(AnthropicModelVariant::Claude3Sonnet20240229),
-            "claude-3-haiku-20240307" => Ok(AnthropicModelVariant::Claude3Haiku20240307),
-            "claude-2.1" => Ok(AnthropicModelVariant::Claude21),
-            "claude-2.0" => Ok(AnthropicModelVariant::Claude20),
-            "claude-instant-1.2" => Ok(AnthropicModelVariant::ClaudeInstant12),
-            _ => Err(OpenAIError::InvalidArgument(format!(
-                "Unknown model variant: {s}"
-            ))),
-        }
-    }
-}
-
-impl Display for AnthropicModelVariant {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let z = match self {
-            AnthropicModelVariant::Claude35Sonnet20240620 => "claude-3-5-sonnet-20240620",
-            AnthropicModelVariant::Claude3Opus20240229 => "claude-3-opus-20240229",
-            AnthropicModelVariant::Claude3Sonnet20240229 => "claude-3-sonnet-20240229",
-            AnthropicModelVariant::Claude3Haiku20240307 => "claude-3-haiku-20240307",
-            AnthropicModelVariant::Claude21 => "claude-2.1",
-            AnthropicModelVariant::Claude20 => "claude-2.0",
-            AnthropicModelVariant::ClaudeInstant12 => "claude-instant-1.2",
-        };
-        write!(f, "{z}")
-    }
-}
-
-impl Serialize for AnthropicModelVariant {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        serializer.serialize_str(&self.to_string())
-    }
-}
-
-// Custom Deserialize implementation to use from_str() for deserialization
-impl<'de> Deserialize<'de> for AnthropicModelVariant {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        let s = String::deserialize(deserializer)?;
-        AnthropicModelVariant::from_str(&s).map_err(serde::de::Error::custom)
-    }
-}
-
-impl AnthropicModelVariant {
-    #[must_use]
-    pub fn default_max_tokens(&self) -> u32 {
-        match self {
-            AnthropicModelVariant::Claude35Sonnet20240620
-            | AnthropicModelVariant::Claude3Opus20240229 => 8192,
-            AnthropicModelVariant::Claude3Sonnet20240229
-            | AnthropicModelVariant::Claude3Haiku20240307
-            | AnthropicModelVariant::Claude21
-            | AnthropicModelVariant::Claude20
-            | AnthropicModelVariant::ClaudeInstant12 => 4096,
-        }
+// Combined pattern that matches all three formats:
+// 1. Anthropic API: claude-3-5-sonnet-20241022 or claude-3-5-sonnet-latest
+// 2. AWS Bedrock: anthropic.claude-3-5-sonnet-20241022-v2:0
+// 3. GCP Vertex AI: claude-3-5-sonnet-v2@20241022
+pub(crate) static ANTHROPIC_REGEX: &str = r"(?x) # Enable verbose mode
```

Comment:
Is there a way to interrogate the anthropic endpoint for the list of models it supports? i.e. a custom anthropic-compatible API might not support all of the official ones. And if a new model is added that doesn't match our existing regex, then we need to release a new version before people can use it.

Reply:
I do not believe any such API exists. The only provider I've seen is xAI.